uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/BaseClass.php

```diff
@@ -0,0 +1,323 @@
+   * @var array
+   */
+  private $leaveImportSuccess = [];
+
+  /**
+   * @var array
+   */
+  private $leaveImportError = [];
+
+  /**
+   * Class constructor.
+   *
+   * @param array $mapperKeys
+   */
+  public function __construct(&$mapperKeys) {
+    parent::__construct();
+    $this->_mapperKeys = &$mapperKeys;
+  }
+
+  /**
+   * Function of undocumented functionality required by the interface.
+   */
+  protected function fini() {}
+
+  /**
+   * The initializer code, called before the processing.
+   */
+  public function init() {
+    $activityContact = CRM_Activity_BAO_ActivityContact::import();
+    $activityTarget['target_contact_id'] = $activityContact['contact_id'];
+
+    $fields = self::getFields();
+    $fields = ['' => ['title' => ts('- do not import -')]] + $fields;
+    foreach ($fields as $name => $field) {
+      $field['type'] = CRM_Utils_Array::value('type', $field);
+      $field['dataPattern'] = CRM_Utils_Array::value('dataPattern', $field, '//');
+      $field['headerPattern'] = CRM_Utils_Array::value('headerPattern', $field, '//');
+      $this->addField($name, $field['title'], $field['type'], $field['headerPattern'], $field['dataPattern']);
+    }
+
+    $this->setActiveFields($this->_mapperKeys);
+    $this->absenceTypes = $this->getAbsenceTypes();
+    $this->absenceStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+  }
+
+  /**
+   * Handle the values in mapField mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   */
+  public function mapField(&$values) {
+    return CRM_Import_Parser::VALID;
+  }
+
+  /**
+   * Handle the values in preview mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   *   the result of this processing
+   */
+  public function preview(&$values) {
+    return $this->summary($values);
+  }
+
+  /**
+   * Handle the values in summary mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   *   the result of this processing
+   */
+  public function summary(&$values) {
+    $erroneousField = NULL;
+    $this->setActiveFieldValues($values, $erroneousField);
+    $params = &$this->getActiveFieldParams();
+    $errorMessage = NULL;
+    $errorMessage .= $this->validateFieldTypes($params);
+    $errorMessage .= $this->validateOptionValues($params);
+
+    if (!empty($errorMessage)) {
+      array_unshift($values, $errorMessage);
+      $errorMessage = NULL;
+      return CRM_Import_Parser::ERROR;
+    }
+
+    return CRM_Import_Parser::VALID;
+  }
+
+  /**
+   * Handle the values in import mode.
+   *
+   * @param int $onDuplicate
+   *   The code for what action to take on duplicates.
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   *   the result of this processing
+   */
+  public function import($onDuplicate, &$values) {
```

Comment:
Can you explain here, in the method docblock (or wherever you think it is better), how the importer handles the records in the CSV file. Especially how it deals with the fact that each row is a separate date of a leave request.